pkg/cloud/services/compute/instance.go

```diff
@@ -258,7 +259,40 @@ func (s *Service) createInstance(eventObject runtime.Object, clusterName string,
 		AccessIPv4:       accessIPv4,
 	}
 
-	serverCreateOpts = applyRootVolume(serverCreateOpts, instanceSpec.RootVolume)
+	volume, err := s.getOrCreateRootVolume(eventObject, instanceSpec, imageID)
+	if err != nil {
+		return nil, fmt.Errorf("error in get or create root volume: %w", err)
+	}
+
+	instanceCreateTimeout := getTimeout("CLUSTER_API_OPENSTACK_INSTANCE_CREATE_TIMEOUT", timeoutInstanceCreate)
+	instanceCreateTimeout *= time.Minute
+
+	// Wait for volume to become available
+	if volume != nil {
+		err = util.PollImmediate(retryIntervalInstanceStatus, instanceCreateTimeout, func() (bool, error) {
```

Comment:
`instanceCreateTimeout` is used but actually this waits for volume to become available so that it is better to introduce new variable. We can follow up this issue after this PR merged because time out settings are not consistently defined: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/33a98520ea942a1e011f249d36d8541c84bbb558/pkg/cloud/services/compute/instance.go#L41-L43

Reply:
Honestly I'd probably just rename the variable to be something like openStackRetryInterval. In practise nobody is ever going to tune these individually, and there would (hopefully!) be very minimal benefit if they did.

However, my grand plan (see my other comment reply!) is to push this polling out to the controller. Blocking in the reconcile loop is potentially problematic anyway.

Reply:
I agree creating instance logic is too big and hard to review now. We really need refactoring.